### PR TITLE
boards: Add support of network interfaces for NRF boards.

### PR DESCRIPTION
**Pull request by katgiadla**
This changes add to files with boards specification information about support of the network interface.

Signed-off-by: Katarzyna Giądła <katarzyna.giadla@nordicsemi.no>

### CODE REVIEW

File: boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160.yaml [hunk 1]
```diff
@@ -19,3 +19,4 @@ supported:
   - spi
   - watchdog
   - counter
+  - netif
```
Comment (rlubos) [inline boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160.yaml:22]: I'm not sure if we want to add the `netif` capability to the secure board definition. In practice, we only build networking for non-secure part.

File: boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpuapp.yaml [hunk 2]
```diff
@@ -16,3 +16,4 @@ supported:
   - watchdog
   - usb_cdc
   - usb_device
+  - netif
```
Comment (rlubos) [inline boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpuapp.yaml:19]: Ditto.

File: boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160_ns.yaml [hunk 3]
```diff
@@ -16,3 +16,4 @@ supported:
   - i2c
   - pwm
   - watchdog
+  - netif
```
Comment (rlubos) [inline boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160_ns.yaml:19]: Could use `netif:modem` if we want to be more specific.

File: boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpunet.yaml [hunk 4]
```diff
@@ -10,3 +10,4 @@ ram: 64
 flash: 256
 supported:
   - watchdog
+  - netif
```
Comment (rlubos) [inline boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpunet.yaml:13]: Afaik, in contrary to the name, we don't do any networking on the net core, it just runs the driver.

File: boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpuapp_ns.yaml [hunk 5]
```diff
@@ -14,3 +14,4 @@ supported:
   - watchdog
   - usb_cdc
   - usb_device
+  - netif
```
Comment (rlubos) [inline boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpuapp_ns.yaml:17]: `netif:opentread`?

File: boards/arm/nrf52dk_nrf52832/nrf52dk_nrf52832.yaml [hunk 6]
```diff
@@ -20,3 +20,4 @@ supported:
   - pwm
   - spi
   - watchdog
+  - netif
```
Comment (rlubos) [inline boards/arm/nrf52dk_nrf52832/nrf52dk_nrf52832.yaml:23]: Do we really want `netif` on this board, it's BLE only?

File: boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpuapp_ns.yaml (hunk 5 above)
Comment (katgiadla) [reply boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpuapp_ns.yaml:17]: Done

File: boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160_ns.yaml (hunk 3 above)
Comment (katgiadla) [reply boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160_ns.yaml:19]: Done

File: boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160.yaml (hunk 1 above)
Comment (rlubos) [reply boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160.yaml:22]: True, but do we really want/need to run tests in the secure domain?

File: boards/arm/nrf52dk_nrf52832/nrf52dk_nrf52832.yaml (hunk 6 above)
Comment (rlubos) [reply boards/arm/nrf52dk_nrf52832/nrf52dk_nrf52832.yaml:23]: I'd opt for removing the capability here.

File: boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160.yaml (hunk 1 above)
Comment (katgiadla) [reply boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160.yaml:22]: Done

File: boards/arm/nrf52dk_nrf52832/nrf52dk_nrf52832.yaml (hunk 6 above)
Comment (katgiadla) [reply boards/arm/nrf52dk_nrf52832/nrf52dk_nrf52832.yaml:23]: Done

Comment (katgiadla) [reply boards/arm/nrf52dk_nrf52832/nrf52dk_nrf52832.yaml]: Done

File: boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpuapp.yaml (hunk 2 above)
Comment (katgiadla) [reply boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpuapp.yaml]: Done

File: boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpunet.yaml (hunk 4 above)
Comment (katgiadla) [reply boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpunet.yaml:13]: Should this capability remove for this board?

**Review by rlubos — DISMISSED** [review]

**Review by rlubos — DISMISSED** [review]

**Review by rlubos — DISMISSED** [review]

File: boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpuapp_ns.yaml [hunk 7]
```diff
@@ -14,3 +14,4 @@ supported:
   - watchdog
   - usb_cdc
   - usb_device
+  - netif:openthread
```
Comment (anangl) [inline boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpuapp_ns.yaml:17]: Why the same is not added in `nrf5340dk_nrf5340_cpuapp.yaml` (similarly for thingy53)?

Comment (katgiadla) [reply boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpuapp_ns.yaml:17]: It was metioned here: https://github.com/zephyrproject-rtos/zephyr/pull/50986#discussion_r987791105

Comment (anangl) [reply boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpuapp_ns.yaml:17]: For nRF9160 - agreed. But for nRF5340, I believe it is a valid use case to use openthread also without secure/non-secure split on the application core. And to be sure that both `nrf5340_cpuapp` targets are properly supported, both should be tested. I discussed it offline with @rlubos and we agreed on that. Thus please, add that capability also for secure targets in `nrf5340dk` and `thingy53`.

**Review by anangl — DISMISSED** [review]

**Review by rlubos — DISMISSED** [review]

**Review by anangl — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**